### ISSUE TITLE
How many epochs recommended for Transfer Learning ?

### ISSUE THREAD

**Issue by jamshaidsohail5** [ultralytics/yolov5#366]
## ❔Question

Hi everyone. I was using YOLO-V5 for fine-tuning on my own dataset using coco-pretrained weights. As there is no paper for YOLO-V5, I was just wondering how many epochs are recommended for good performance? I was able to get the optimal batch-size value from the main page of the repo. Any help would be highly appreciated.

**Comment by glenn-jocher** [comment]
@jamshaidsohail5 train with all default settings (300 epochs), both from scratch and then from pretrained weights before worrying about changing the defaults.

**Comment by marvision-ai** [comment]
@glenn-jocher for finetuning on your own dataset, I feel like 300 epochs is too many. Am I incorrect?

I understand training from scratch for 300, but can you elaborate a bit more on the finetuning part? usually I get a good accuracy after ~30 epochs using the pre-trained coco models from gDrive.

**Comment by glenn-jocher** [comment]
@marvision-ai there are no blanket statements here as all datasets are different, but in general if you are not observing overfitting (rising val losses) you have not trained long enough.

For every 10 people that train too little there will only be 1 or 2 that train too long.

**Comment by marvision-ai** [comment]
@glenn-jocher okay cool! I have a pretty big project in the works and I'll update you with results in the coming days. Thanks!

**Comment by siddagra** [comment]
Even with 300 epochs, I never see my val loss increasing. 
Isn't the specific point of OneCycleLR such that it will increase val loss initially but then ramp down? Is the final_LR for OneCycleLR set too high? 

Also, would really appreciate TPU support as this takes weeks to train even the smallest model (mostly CPU being the bottleneck).

**Comment by glenn-jocher** [comment]
@siddagra Impressive! It sounds like your training system is well optimized. Regarding OneCycleLR, by design, it’s normal to observe an initial increase in validation loss, followed by a decreasing trend. The final_LR for an epoch should technically not be too high if validation loss keeps decreasing.

Regarding TPU support, this is not currently available but GPU support is provided. We are constantly improving our system, and your feedback is valuable. Thank you for your understanding.
